dashboard/app/controllers/api/v1/ml_models_controller.rb

```diff
@@ -65,6 +65,9 @@ def names
   # GET api/v1/ml_models/:id
   # Retrieve a trained ML model from S3
   def show
+    # Before attempting to fetch a model from s3, check that the id param
+    # matches the expected format.
+    return head :not_found unless UserMlModel.valid_model_id?(params[:id])
```

Comment:
ah, sorry for the piecemeal review, but i just realized that this controller generates the UserMlModel model_id -- that should be a class method on the model as well, but doesn't necessarily need to go in this PR